MapboxCoreNavigation/RouteController.swift

```diff
@@ -1,199 +1,153 @@
     
-    /**
-     If true, the `RouteController` attempts to calculate a more optimal route for the user on an interval defined by `RouteControllerProactiveReroutingInterval`.
-     */
-    @objc public var reroutesProactively = false
-
-    var didFindFasterRoute = false
-
-    /**
-     Details about the user’s progress along the current route, leg, and step.
-     */
-    @objc public var routeProgress: RouteProgress {
+    public var route: Route {
         get {
-            return _routeProgress
+            return routeProgress.route
         }
         set {
-            if let location = self.location {
-                delegate?.router?(self, willRerouteFrom: location)
-            }
-            _routeProgress = newValue
-            announce(reroute: routeProgress.route, at: dataSource.location, proactive: didFindFasterRoute)
+            routeProgress = RouteProgress(route: newValue)
+            updateNavigator()
         }
     }
     
     private var _routeProgress: RouteProgress {
         didSet {
             movementsAwayFromRoute = 0
+            updateNavigator()
         }
     }
     
-    public var route: Route {
-        get {
-            return routeProgress.route
-        }
-        set {
-            routeProgress = RouteProgress(route: newValue)
-        }
-    }
-
-    var isRerouting = false
-    var lastRerouteLocation: CLLocation?
-
-    var routeTask: URLSessionDataTask?
-    var lastLocationDate: Date?
-
-    var hasFoundOneQualifiedLocation = false
-
     var movementsAwayFromRoute = 0
-
-    var previousArrivalWaypoint: Waypoint?
-
+    
+    var routeTask: URLSessionDataTask?
+    
+    var lastRerouteLocation: CLLocation?
+    
+    var didFindFasterRoute = false
+    
+    var isRerouting = false
+    
     var userSnapToStepDistanceFromManeuver: CLLocationDistance?
     
+    var previousArrivalWaypoint: Waypoint?
+    
+    var isFirstLocation: Bool = true
+    
     /**
-     Intializes a new `RouteController`.
-
-     - parameter route: The route to follow.
-     - parameter directions: The Directions object that created `route`.
-     - parameter source: The data source for the RouteController.
+     The threshold used when we determine when the user has arrived at the waypoint.
+     By default, we claim arrival 5 seconds before the user is physically estimated to arrive.
      */
-    required public init(along route: Route, directions: Directions = Directions.shared, dataSource source: RouterDataSource) {
-        self.directions = directions
-        self._routeProgress = RouteProgress(route: route)
-        self.dataSource = source
-        UIDevice.current.isBatteryMonitoringEnabled = true
-
-        super.init()
-        
-        checkForUpdates()
-        checkForLocationUsageDescription()
+    @objc func setWaypointArrivalThreshold(_ threshold: TimeInterval) {
+        navigator.configureNavigator(forName: "arrival_threshold_duration", value: "\(threshold)")
```

Comment:
This is less than ideal, however, the property is currently write-only in nav-native.

Reply:
Unless MapboxNavigationNative reserves the right to silently change the threshold, we could store a copy of the new value to be returned in the getter.